File: app/src/main/java/io/mgba/Model/Interfaces/ILibrary.java
package io.mgba.Model.Interfaces;

import java.util.List;

import io.mgba.Data.Database.Game;
import io.reactivex.Single;

public interface ILibrary {
    Single<List<Game>> prepareGames(int platform);
    Single<List<Game>> query(String query);

    Single<List<Game>> reloadGames(int... platform);
    Single<List<Game>> reloadGames(String path, int... platform);
}

File: app/src/main/java/io/mgba/model/interfaces/ILibrary.kt
package io.mgba.model.interfaces

import io.mgba.data.database.Game
import io.reactivex.Single

interface ILibrary {
    fun prepareGames(platform: Int): Single<List<Game>>
    fun query(query: String): Single<List<Game>>

    fun reloadGames(vararg platform: Int): Single<List<Game>>
    fun reloadGames(path: String, vararg platform: Int): Single<List<Game>>
}
